docs/new_protocol.asciidoc

```diff
@@ -0,0 +1,463 @@
+file on Github.
+
+==== Cloning and compiling
+
+After having https://golang.org/doc/install[installed Go] and having setup the
+https://golang.org/doc/code.html#GOPATH[GOPATH] environment variable to point to
+your preferred workspace location, you can clone and compile Packetbeat with the
+following commands:
+
+[source,shell]
+----------------------------------------------------------------------
+$ mkdir -p $GOPATH/src/github.com/elastic
+$ cd $GOPATH/src/github.com/elastic
+$ git clone https://github.com/elastic/packetbeat.git
+----------------------------------------------------------------------
+
+and then compile it with:
+
+[source,shell]
+----------------------------------------------------------------------
+$ cd packetbeat
+$ make
+----------------------------------------------------------------------
+
+Note that the location where you clone is important. If you prefer working
+outside of the `GOPATH` environment, you can clone to another directory and only
+create a symlink it to the `$GOPATH/src/github.com/elastic/` directory.
+
+==== Fork and branch
+
+We recommend the following work flow for contributing to Packetbeat:
+
+* Fork Packetbeat in Github to your own account
+
+* In the `$GOPATH/src/github.com/elastic/packetbeat` folder, add your fork
+  as a new remote. For example (replace `tsg` with your github account):
+
+[source,shell]
+----------------------------------------------------------------------
+git remote add tsg git@github.com:tsg/packetbeat.git
+----------------------------------------------------------------------
+
+* Create a new branch for your work:
+
+[source,shell]
+----------------------------------------------------------------------
+git checkout -b cool_new_protocol
+----------------------------------------------------------------------
+
+* Commit as often as you like, push to your private fork with:
+
+[source,shell]
+----------------------------------------------------------------------
+git push --set-upstream tsg cool_new_protocol
+----------------------------------------------------------------------
+
+* When you are ready to submit your PR, simply do so from the GitHub web
+  interface. Feel free to submit your PR early, you can still add commits to
+  the branch after creating the PR. Submitting it early gives us more time to
+  provide feedback and perhaps help you with it.
+
+==== A note about dependencies
+
+Packetbeat uses https://github.com/tools/godep[Godep] for vendoring its
+dependencies. The way Godep works is that it copies the source code of all
+dependencies in the `Godep/_workspace` folder.
+
+If you add a new dependency, you can use `go get` normally, and then run the
+following command to add the new dependency to `Godep`.
+
+[source,shell]
+----------------------------------------------------------------------
+godep update ...
+----------------------------------------------------------------------
+
+=== Protocol modules
```

Comment:
Missing:
  - add config to packetbeat/config.go (to be read by yaml modul)
  - add module to main.go so it will be registered
  - document new fields in etc/fields.py, as fields.py is used by integration tests to test for mandatory/unknown fields
  - (optional) describe protocol configuration in etc/configuration.ascii
  - (optional) update default config files: etc/packetbeat.yml, packetbeat.dev.yml and samples in getting started 
  - (optional) make gen after updating fields.py and getting started doc to update the generated documentation